src/main/java/net/minecrell/serverlistplus/core/player/ban/BanDetector.java

```diff
@@ -0,0 +1,44 @@
+package net.minecrell.serverlistplus.core.player.ban;
+
+import net.minecrell.serverlistplus.core.player.PlayerIdentity;
+
+import java.sql.Timestamp;
+
+public interface BanDetector {
```

Comment:
Maybe `BanProvider` would be a better name?

Reply:
I agree. I was unsure what to name it. But I like that name.